embassy-rp/src/uart/mod.rs

```diff
@@ -369,11 +350,7 @@ impl<'d, T: Instance, M: Mode> Uart<'d, T, M> {
                 pin.pad_ctrl().write(|w| w.set_ie(true));
             }
 
-	    let (baud_ibrd, baud_fbrd) = Uart::<T,M>::baudrate_calculations(config.baudrate);
-
-            // Load PL011's baud divisor registers
-            r.uartibrd().write_value(pac::uart::regs::Uartibrd(baud_ibrd));
-            r.uartfbrd().write_value(pac::uart::regs::Uartfbrd(baud_fbrd));
+	    Uart::<T,M>::set_baudrate_inner(config.baudrate);
```

Comment:
It's nicer to do `Self::set_baudrate_inner`, to not repeat the generic params.

Reply:
Sure thing boss just a sec

Reply:
I know but the compiler complaints about not allowing Generics over Self unless I add a lifetime specifier which I'm not confident could alter many other stuff inside the crate itself. Either way I send it through.

Just to let you know here is the error I was telling, in case you want to know.

```
cargo check
    Checking embassy-rp v0.1.0 (/home/andresh/Projects/embassy/embassy-rp)
error[E0109]: type arguments are not allowed on self type
   --> src/uart/mod.rs:353:13
    |
353 |         Self::<T,M>::set_baudrate_inner(config.baudrate);
    |         ----   ^ ^ type argument not allowed
    |         |
    |         not allowed on self type
    |
note: `Self` is of type `uart::Uart<'d, T, M>`
   --> src/uart/mod.rs:85:12
    |
85  | pub struct Uart<'d, T: Instance, M: Mode> {
    |            ^^^^ `Self` corresponds to this type
...
302 | impl<'d, T: Instance, M: Mode> Uart<'d, T, M> {
    | --------------------------------------------- `Self` is on type `Uart` in this `impl`
help: the `Self` type doesn't accept type parameters, use the concrete type's name `Uart` instead if you want to specify its type parameters
    |
353 |         Uart::<T,M>::set_baudrate_inner(config.baudrate);
    |         ~~~~

```

Reply:
The Self is there are we good now ?

Reply:
> LGTM. Can you fix the CI failures? (run `cargo fmt`. I recommend setting your IDE to do that automatically on save)

Sorry mate, gave it a try but the CI seems to be failing on a thing that is not displayed on the log. The only thing that shows is:

1) the curl pulling the batch
2) the setup of the toolchain
3) A long diff with the changes (I guess)
3) a plain vanilla error message without any references or explaination. that says

`
Error: Process completed with exit code 123.
`
And there isn't any thing further. 

Looking at eh `ci.sh` script,  looks like is some stuff related with the `run_elf` function which is for testing on the probes I guess, that is out of my reach unless you teach me.

Reply:
Failure is due to not having formatted your code with rustfmt.

Reply:
Alright mate Done. 

You live you learn !!!!!

First time for me on a PR